gprofiler/profilers/java.py

```diff
@@ -259,11 +288,71 @@ def make_application_metadata(self, process: Process) -> Dict[str, Any]:
         # that loads other libs.
         libjvm_elfid = get_mapped_dso_elf_id(process, "/libjvm")
 
-        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid}
+        metadata = {
+            "java_version": version,
+            "libjvm_elfid": libjvm_elfid,
+            "jvm_flags": self.get_jvm_flags_serialized(process),
+        }
 
         metadata.update(super().make_application_metadata(process))
         return metadata
 
+    def get_jvm_flags_serialized(self, process: Process) -> Optional[List[Dict]]:
+        jvm_flags = self.get_jvm_flags(process)
+        if jvm_flags is None:
+            return None
+
+        return [flag.to_dict() for flag in sorted(jvm_flags, key=lambda flag: flag.name)]
+
+    def get_jvm_flags(self, process: Process) -> Optional[List[JvmFlag]]:
+        jvm_raw_flags: Optional[List[JvmFlag]] = self.get_jvm_flags_raw(process)
+
+        if jvm_raw_flags is None:
+            return None
+
+        elif self.jvm_flags_to_retrieve == JavaFlagRetrievalOptions.ALL:
+            return jvm_raw_flags
+        elif self.jvm_flags_to_retrieve == JavaFlagRetrievalOptions.DEFAULT:
+            return [flag for flag in jvm_raw_flags if self.filter_jvm_flag(flag)]
+        elif isinstance(self.jvm_flags_to_retrieve, list):
+            requested_flags = [flag for flag in jvm_raw_flags if flag.name in self.jvm_flags_to_retrieve]
+            not_found_requested_flag_names = [
+                flag_name
+                for flag_name in self.jvm_flags_to_retrieve
+                if flag_name not in [flag.name for flag in requested_flags]
+            ]
+
+            logger.error("Flags not found in jvm flags: %s", not_found_requested_flag_names)
+
+            return requested_flags
+
+        logging.error("Unrecognized jvm_flags_to_retrieve value: %s", self.jvm_flags_to_retrieve)
+
+        return None
+
+    @staticmethod
+    def filter_jvm_flag(flag: JvmFlag) -> bool:
```

Comment:
Document the decisions in this function.

"default" we skip because - we don't want (?)

"ccstrlist" we skip because not implemented

"notproduct" etc we skip because it's debugging flags (?)